otter/scheduler.py

```diff
@@ -19,81 +19,47 @@
 from otter.util.hashkey import generate_transaction_id
 
 
-class SchedulerService(TimerService):
+class SchedulerService(MultiService, object):
     """
     Service to trigger scheduled events
     """
 
-    def __init__(self, batchsize, interval, store, kz_client,
-                 zk_partition_path, time_boundary, buckets, clock=None, threshold=60):
+    def __init__(self, batchsize, store, partitioner_factory, threshold=60):
         """
         Initialize the scheduler service
 
         :param int batchsize: number of events to fetch on each iteration
-        :param int interval: time between each iteration
-        :param kz_client: `TxKazooClient` instance
-        :param buckets: an iterable containing the buckets which contains scheduled events
-        :param zk_partition_path: Partiton path used by kz_client to partition the buckets
-        :param time_boundary: Time to wait for partition to become stable
-        :param clock: An instance of IReactorTime provider that defaults to reactor if not provided
+        :param store: cassandra store
+        :param partitioner_factory: Callable of (log, callback) ->
+            :obj:`Partitioner`
         """
-        TimerService.__init__(self, interval, self.check_events, batchsize)
+        super(SchedulerService, self).__init__()
```

Comment:
(Notably, see how `object` appears *after* `MultiService`, which is the special case in which maybe sometimes it works.)